storage/innobase/handler/handler0alter.cc

```diff
@@ -5886,6 +5886,21 @@ ha_innobase::commit_inplace_alter_table(
 	row_mysql_unlock_data_dictionary(trx);
 	trx_free_for_mysql(trx);
 
+	/* Rebuild index translation table now for temporary tables if we are
+	restoring secondary keys, as ha_innobase::open will not be called for
+	the next access.  */
+	if (DICT_TF2_FLAG_IS_SET(ctx0->new_table, DICT_TF2_TEMPORARY)
```

Comment:
Can we replace this with dict_table_is_temporary() to make it more readable and consistent with 5.5 too.